src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/BlockListener.java

```diff
@@ -58,7 +61,18 @@ public void onBlockPlaceExisting(BlockPlaceEvent e) {
         // This prevents Players from placing a block where another block already exists
         // While this can cause ghost blocks it also prevents them from replacing grass
         // or saplings etc...
-        if (BlockStorage.hasBlockInfo(e.getBlock())) {
+        Block block = e.getBlock();
+        if (BlockStorage.hasBlockInfo(block)) {
+            if (e.getBlockReplacedState().getType().isAir()) {
+                Iterator<ItemStack> item = BlockStorage.check(block).getDrops().iterator();
```

Comment:
Calling `BlockStorage.check()` here is completely redundant.
You are already doing a check by doing `hasBlockInfo()` above, this is essentially doing two concurrent operations redundantly, a null check would be more sufficient here.